azurerm/resource_arm_automation_schedule.go

```diff
@@ -257,6 +336,9 @@ func resourceArmAutomationScheduleRead(d *schema.ResourceData, meta interface{})
 	if v := resp.TimeZone; v != nil {
 		d.Set("timezone", v)
 	}
+	if v := resp.AdvancedSchedule; v != nil {
+		d.Set("advanced_schedule", flattenArmAutomationScheduleAdvanced(v))
+	}
```

Comment:
can we ensure we set this all the time and move the nil-check within the flatten function? that ensures changes to this are detected. in addition - given this is a complex object can we check for errors when setting it? e.g.

```
if err := d.Set("advanced_schedule", flattenArmAutomationScheduleAdvanced(v)); err != nil {
  return fmt.Errorf("Error setting `advanced_schedule`: %+v", err)
}
```